### ISSUE TITLE
Allow defining aliases for channels

### ISSUE THREAD

**Issue by S-S-X** [mt-mods/beerchat#56]
Longer discussion about this and why it would be useful at pandorabox #mod channel.
Not going to replay or parse that discussion here, instead ask if more details needed or something is not clear.

### What is needed:
* Command to add alias for channel, possibly also ability to convert existing channel to alias while retaining subscribers.
* Translation for channel aliases before handing message for plugin / default message processor (possibly on_receive hook).
* Handling for `/cc`, `/jc` and `/dc` commands, these should have their own event callbacks to hand full responsibility to plugins without cluttering core functions. Note that solving #54 could help a lot with this.

Everything should be implemented either in separate plugin like `/plugin/hash_alias.lua` or by implementing functionality directly in `/plugin/hash.lua`.

### How channel aliases would be handled:
Assume that there's channels `#barter`, `#trade` and `#ads` all created to serve exact same or very much overlapping purpose. All channels are getting fairly low traffic.
You'd like to combine all those but allow players to continue using channel they were using before, also you'd like to ensure that someone who does not know about those channels wont accidentally get idea that new channel is needed efficiently dividing already users and confusing existing users.

1. One channel acts as master channel, that will be real chat channel. We're going to use `#trade` for this example.
2. Other channels will act as aliases for `#trade` channel, in this case aliases are `#barter` and `#ads`.
3. Command `/chat_channel_alias <target> <alias> [<alias> ...]` to link channels, example: `/chat_channel_alias #trade #barter #ads`.
4. Player Sam uses `/jc #ads`, he will be joined to `#trade` channel.
5. Player SX uses `/jc #trade`, he will be joined to `#trade` channel.
6. Player Sam uses `#ads Hello there o/`, both players will see message on `#trade` channel.
7. Player SX uses `#trade Hello there o/`, both players will see message on `#trade` channel.
8. Player Sam uses `#barter Hello there o/`, both players will see message on `#trade` channel.
9. Player Sam uses `#barter`, default channel for Sam is set to `#trade` channel.

### Conclusion:

Basically that explains probably simplest solution: simply translate channel names based on alias list which is hash table and therefore instant lookup. Translate channel name before applying other message handlers, only alias plugin will ever see aliases and everything else will always see target.
No changes required at any other place, completely pluggable solution and fairly cheap in terms of processing power.
